doc/website/concepts/architecture.md

```diff
@@ -1 +1,64 @@
 # Architecture
+
+This article provides an overview of the Eclipse iceoryx architecture and explains the rationale for it.
+
+## Software layers
+
+The main packages which Eclipse iceoryx is composed of are depicted below.
+
+![Software layers](iceoryx_software_layers_v2_0_0.svg)
+
+The next sections briefly describe the components and their libraries one by one.
+
+## Components and libraries
+
+The different libraries and their namespaces are depicted below.
+
+![Component diagram](iceoryx_components_diagram_v2_0_0.svg)
+
+### iceoryx hoofs
+
+Handy Objects Optimized For Safety (hoofs) is a library and contains various building blocks like fixed size containers,
+concurrency classes and modern, next-gen C++ constructs from upcoming C++ standard releases.
+
+For more information about the components, refer to its [detailed description](iceoryx_hoofs.md).
```

Comment:
Link broken.

Reply:
See my [comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1203#discussion_r818093610). If we use a relative link here, we would need to adapt it, if this file is moved. If we let `mkdocs-autolinks-plugin` do the magic, we don't ever have to care about this link again unless the file name changes.

I think most users access iceoryx.io via the website and don't read it via the GitHub web GUI reader. If you want I can dig out the click rates for both the iceoryx.io and the related markdown files on GitHub, so we can compare them.

Reply:
I personally cannot say what I prefer here. But I have to admit that relative links (or absolute ones) become unmaintainable fast.

Depends on our goals I think, but I do think we should strive to keep the maintenance effort for documentation low (wrt. moving files etc.). If a file is moved we would have to track all files linking it otherwise, which is cumbersome.

Just my 2 cents, others are certainly more knowledgeable than myself in these matters and hence I leave this to them. I just would like it to be as easy as possible to add (technical)  documentation.

Reply:
Ok, I changed the links relatives ones. I still think we'll (again) end up having maintenance problems, but want to unblock this PR. And continue the discussion after the `v2.0` release.